.travis/script.sh

```diff
@@ -42,7 +42,7 @@ python --version || die
     export CPPFLAGS="-DMBEDTLS_DEBUG_C"
     export CPPFLAGS="${CPPFLAGS} -I${TRAVIS_BUILD_DIR}/third_party/mbedtls"
     export CPPFLAGS="${CPPFLAGS} -I${TRAVIS_BUILD_DIR}/third_party/mbedtls/repo/include"
-    export CPPFLAGS="${CPPFLAGS} -DMBEDTLS_CONFIG_FILE=\\\"mbedtls-config.h\\\""
+    export CPPFLAGS="${CPPFLAGS} -DMBEDTLS_CONFIG_FILE=\\\"ot-mbedtls-default-config.h\\\""
```

Comment:
Can you provide some background for the file name change?

Reply:
mbedtls-config.h is too common as a default header file provided by OpenThread. By adding a prefix, users may use the common name when they provide their own mbedtls config file. Besides, the one we provided depends on core config so it cannot be a public header which is included by mbedtls's public headers.

Reply:
> mbedtls-config.h is too common as a default header file provided by OpenThread. By adding a prefix, users may use the common name when they provide their own mbedtls config file.

In the case of a user providing their own mbedtls config, we should remove `third_party/mbedtls` from the include path.

> Besides, the one we provided depends on core config so it cannot be a public header which is included by mbedtls's public headers.

Yes, `third_party/mbedtls/mbedtls-config.h` should only be used if mbedTLS is being built only for OpenThread.

Reply:
reverted.